owl/SOMA-IO.owl

```diff
@@ -25,40 +124,321 @@
     
 
 
-    <!-- http://www.ease-crc.org/ont/SOMA.owl#ComputationalAgent -->
+    <!-- http://www.ease-crc.org/ont/SOMA.owl#API_Specification -->
+
+    <owl:Class rdf:about="http://www.ease-crc.org/ont/SOMA.owl#API_Specification">
+        <rdfs:subClassOf rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#InterfaceSpecification"/>
+        <rdfs:label xml:lang="en">API Specification</rdfs:label>
+    </owl:Class>
+    
+
+
+    <!-- http://www.ease-crc.org/ont/SOMA.owl#Algorithm -->
+
+    <owl:Class rdf:about="http://www.ease-crc.org/ont/SOMA.owl#Algorithm">
+        <rdfs:subClassOf rdf:resource="http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#Plan"/>
+        <rdfs:comment xml:lang="en">An Algorithm is a finite sequence of well-defined instructions, typically used to solve a class of specific problems or to perform a computation.
+
+From Wikipedia: https://en.wikipedia.org/wiki/Algorithm</rdfs:comment>
+        <rdfs:label xml:lang="en">Algorithm</rdfs:label>
+    </owl:Class>
+    
+
+
+    <!-- http://www.ease-crc.org/ont/SOMA.owl#Archive -->
+
+    <owl:Class rdf:about="http://www.ease-crc.org/ont/SOMA.owl#Archive">
+        <owl:equivalentClass>
+            <owl:Class>
+                <owl:intersectionOf rdf:parseType="Collection">
+                    <rdf:Description rdf:about="http://www.ease-crc.org/ont/SOMA.owl#Structured_Text"/>
+                    <owl:Restriction>
+                        <owl:onProperty rdf:resource="http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#expresses"/>
+                        <owl:someValuesFrom rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#FileConfiguration"/>
+                    </owl:Restriction>
+                </owl:intersectionOf>
+            </owl:Class>
+        </owl:equivalentClass>
+        <owl:equivalentClass>
+            <owl:Restriction>
+                <owl:onProperty rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#isGivenMeaningBy"/>
+                <owl:someValuesFrom rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#ArchiveFormat"/>
+            </owl:Restriction>
+        </owl:equivalentClass>
+    </owl:Class>
```

Comment:
Since Archive can mean many things, perhaps make the name a little longer and clearer as to which of those meanings is intended.